### ISSUE TITLE
Project isn't compatible with `pnpm >= 8`

### ISSUE THREAD

**Issue by cyber-bloke** [spacedriveapp/spacedrive#729]
### Describe the bug
```
sd/desktop:build: 
@sd/desktop:build: Caused by:
@sd/desktop:build: 
@sd/desktop:build:   process didn't exit successfully: `/Users/pritpal/GitHub/spacedrive/target/release/build/sd-desktop-macos-562bb7fb12232c4f/build-script-build` (exit status: 101)
@sd/desktop:build:   --- stdout
@sd/desktop:build:   cargo:rustc-link-search=native=/Library/Developer/CommandLineTools/usr/lib/swift/macosx
@sd/desktop:build:   cargo:rustc-link-search=native=/usr/lib/swift
@sd/desktop:build:   cargo:rustc-link-lib=clang_rt.osx
@sd/desktop:build:   cargo:rustc-link-search=/Library/Developer/CommandLineTools/usr/lib/clang/14.0.3/lib/darwin
@sd/desktop:build: 
@sd/desktop:build:   --- stderr
@sd/desktop:build:   error: terminated(1): /usr/bin/xcrun --sdk macosx --show-sdk-platform-path output:
@sd/desktop:build:       xcrun: error: unable to lookup item 'PlatformPath' from command line tools installation
@sd/desktop:build:       xcrun: error: unable to lookup item 'PlatformPath' in SDK '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk'
@sd/desktop:build:       
@sd/desktop:build:   thread 'main' panicked at 'Failed to compile swift package sd-desktop-macos', /Users/pritpal/.cargo/registry/src/github.com-1ecc6299db9ec823/swift-rs-1.0.5/src-rs/build.rs:271:17
@sd/desktop:build:   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
@sd/desktop:build: warning: build failed, waiting for other jobs to finish...
@sd/desktop:build: 
@sd/desktop:build: error: failed to run custom build command for `spacedrive v0.1.0 (/Users/pritpal/GitHub/spacedrive/apps/desktop/src-tauri)`
@sd/desktop:build: 
@sd/desktop:build: Caused by:
@sd/desktop:build:   process didn't exit successfully: `/Users/pritpal/GitHub/spacedrive/target/release/build/spacedrive-953885c04bd08641/build-script-build` (exit status: 1)
@sd/desktop:build:   --- stdout
@sd/desktop:build:   cargo:rerun-if-env-changed=TAURI_CONFIG
@sd/desktop:build:   cargo:rerun-if-changed=tauri.conf.json
@sd/desktop:build:   cargo:rustc-cfg=desktop
@sd/desktop:build:   This manifest requires workspace inheritance, but `inherit_workspace` hasn't been called yet
@sd/desktop:build: 
@sd/desktop:build:        Error failed to build app: failed to build app
@sd/desktop:build: 
@sd/desktop:build:  ELIFECYCLE  Command failed with exit code 1.
@sd/desktop:build: 
@sd/desktop:build: Child exited with code 1

 Tasks:    7 successful, 7 total
Cached:    7 cached, 7 total
  Time:    623ms >>> FULL TURBO
```

### Reproduction

git clone  spacedrive
pnpm build
and the error occurs

### Expected behavior

this occurs everytime, i have cleaned and reconfigured it .

### Platform and versions

```Shell
MaCOS Ventura 13.3.1
8.3.1
cargo 1.68.1 (6feb7c9cf 2023-03-26)
rustc 1.68.2 (9eb3afe9e 2023-03-27) (built from a source tarball)

for pnpm install:
Your pnpm version is incompatible with "/tsconfck/2.0.2(typescript@4.9.4)".

Expected version: ^7.18.0
Got: 8.3.1

This is happening because the package's manifest has an engines.pnpm field specified.
To fix this issue, install the required pnpm version globally.

To install the latest version of pnpm, run "pnpm i -g pnpm".
To check your pnpm version, run "pnpm -v".
```

### Stack trace

```Shell
none
```

### Additional context

please let me know what is wrong in my configuration

**Comment by HeavenVolkoff** [comment]
This is a known issue. One of our dependencies needs to be updated to be compatible with `pnpm >= 8`. However, we are currently completely focused on working towards our alpha release, so fixing this issue is a low priority at the moment. In the meantime, you can downgrade to `pnpm 7` and it should work.

**Comment by cyber-bloke** [comment]
Thank you very much. If you could let me know where it is called from, I can try to work around it… Sent from Proton Mail for iOS On Thu, Apr 20, 2023 at 8:29 PM, Vítor Vasconcellos ***@***.***> wrote:  
This is a known issue. One of our dependencies needs to be updated to be compatible with pnpm >= 8. However, we are currently completely focused on working towards our alpha release, so fixing this issue is a low priority at the moment. In the meantime, you can downgrade to pnpm 7 and it should work.

**Comment by Brendonovich** [comment]
Are we sure this a problem with `pnpm >= 8`? I've been using `8.4.0` with no issue. 
Looks to me like `swift-rs` is failing to run `swift build` and Tauri for some reason is complaining about workspace inheritance.

**Comment by brxken128** [comment]
It's probably fine now, there was an issue when pnpm 8 _just_ dropped though.

The Tauri workspace inheritance bug somehow did not hit our app, but it hits every other app with a similar workspace layout We got lucky - myself and @oscartbeaumont have no clue on how it's still working in main.

I think(?) Tauri v1.3.0 or so allows workspace inheritance, so it may be worth upgrading just to avoid this issue altogether.

I believe @fogodev has been using pnpm 8 since it came out, so maybe he has some more insight (we've discussed it in the past but I can't remember off the top of my head).
